src-java/blue-green/src/main/java/org/openkilda/bluegreen/kafka/interceptors/VersioningProducerInterceptor.java

```diff
@@ -0,0 +1,96 @@
+/* Copyright 2020 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.bluegreen.kafka.interceptors;
+
+import static org.openkilda.bluegreen.kafka.Utils.MESSAGE_VERSION_HEADER;
+import static org.openkilda.bluegreen.kafka.Utils.PRODUCER_COMPONENT_NAME_PROPERTY;
+import static org.openkilda.bluegreen.kafka.Utils.PRODUCER_RUN_ID_PROPERTY;
+import static org.openkilda.bluegreen.kafka.Utils.PRODUCER_ZOOKEEPER_CONNECTION_STRING_PROPERTY;
+import static org.openkilda.bluegreen.kafka.Utils.getValue;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.producer.ProducerInterceptor;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+import java.time.Instant;
+import java.util.Map;
+
+@Slf4j
+public class VersioningProducerInterceptor<K, V> extends VersioningInterceptorBase
+        implements ProducerInterceptor<K, V> {
+
+    public VersioningProducerInterceptor() {
+        log.debug("Initializing VersioningProducerInterceptor");
+    }
+
+    @Override
+    public ProducerRecord<K, V> onSend(ProducerRecord<K, V> record) {
+        if (watchDog == null) {
+            // try to reconnect
+            initWatchDog();
```

Comment:
You do not check actual connect results. Is it ok? Should we have something like `isVersionUsable()` and use it instead of null comparison? It can take into account ZK connection status too.

Reply:
Check of connection is done here 
https://github.com/telstra/open-kilda/blob/326f679bb39b26e0975de85babd0fa80d3121f77/src-java/blue-green/src/main/java/org/openkilda/bluegreen/ZkClient.java#L62

But it's good idea to add check here too